sdk/monitor/azure-monitor-query/azure/monitor/query/_log_query_client.py

```diff
@@ -162,9 +162,11 @@ def batch_query(self, queries, **kwargs):
             request_order = [req['id'] for req in queries]
         batch = BatchRequest(requests=queries)
         generated = self._query_op.batch(batch, **kwargs)
-        return LogsBatchResults._from_generated( # pylint: disable=protected-access
-            generated, request_order
-            )
+        return order_results(
+            request_order,
```

Comment:
what if the id in request_order does not exist in results?

Reply:
it will always exist - it's a 1:1 mapping